### ISSUE TITLE
Formatting of nested -ifdef/-ifndef macro directives

### ISSUE THREAD

**Issue by TD5** [WhatsApp/erlfmt#345]
Sometimes I need to nest macro directives, and when I do so, it's nice to indent them as one would do with code generally - just to keep it readable - however, `erlfmt` seems to want to remove all indentation, for example:

-----------------------------

```
-ifdef(FOO).
  -ifndef(BAR).
    -define(BAZ, true).
  -endif.
-endif.
```

`--[apply erlfmt]-->`

```
-ifdef(FOO).
-ifndef(BAR).
-define(BAZ, true).
-endif.
-endif.
```

-----------------------------

I don't think there's any mention this is the style guide. As far as I can see, the macro-related stuff covers various topics, but doesn't consider flow control for macro definitions. Also, I am not a frequent user of `erlfmt`, so maybe I am doing something wrong, making this a support request rather than a feature request.

I know nesting macro directives like this is probably Not Good™, but I'd prefer not to compound that by needing to have them flattened out.

Thanks!

**Comment by michalmuskala** [comment]
This is a curious case, because of how inconsistent this is handled in source code.

For some cases, like the one you listed, the indentation is sometimes applied.

Yet for others, way more common, it almost never is. For example, I don't think I've ever saw code like:

```erlang
-ifdef(FOO)
    conditional_function() ->
        ....
-endif.
```

So I think the primarily challenge of a feature like this would be to decide when to apply it, and when not to.

**Comment by TD5** [comment]
Yes, that's my experience too. My personal feeling, and what I saw in code, was that at one layer of nesting, there's no indentation, but beyond that there is:

One level of macro flow-control:
```
-ifdef(FOO)
conditional_function() ->
    ....
-endif.
```

More than one level of macro flow-control:
```
-ifdef(FOO)
    -ifdef(FOO)
        conditional_function() ->
            ....
    -endif.
-endif.
```
